library/alloc/src/sync.rs

```diff
@@ -3,6 +3,10 @@
 //! Thread-safe reference-counting pointers.
 //!
 //! See the [`Arc<T>`][Arc] documentation for more details.
+//!
+//! **Note**: This module is only available on platforms that support atomic
+//! loads and stores of pointers. This may be detected at compile time using
+//! `#[cfg(target_has_atomic = "ptr")]`.
```

Comment:
Hm, this might be true today, but in the future I could imagine that we'd have some parts of this module available even without atomics of pointer width.

It's already a bit weird that alloc::sync is not a superset of core::sync, it seems; that seems unusual.

Reply:
> Hm, this might be true today, but in the future I could imagine that we'd have some parts of this module available even without atomics of pointer width.

The documentation could then be revised at the same time the `cfg` is removed.

> It's already a bit weird that alloc::sync is not a superset of core::sync, it seems; that seems unusual.

That sounds like something worth fixing for consistency, but this change is forward-compatible with that — the note on the modules would just be removed, and the note on `Arc` kept. (`alloc::task` is in the same situation — and so technically the `Wake` trait should have a similar note.)